### DISCUSSION TITLE
FlexSearch not functioning

### DISCUSSION THREAD

**Discussion by krzko** [thuliteio/doks#310] [Q&A] [answered]
Greetings,

Have a play with **doks** and trying to get FlexSearch to work for our pages. Doesn't seem to return any results. It's an english site, so have added the latin values for `encode` and `tokenize`, as per;

```js
var index = new FlexSearch({
    preset: 'score',
    cache: true,
    doc: {
        id: 'id',
        field: [
          'title',
          'description',
          'content',
        ],
        store: [
          'href',
          'title',
          'description',
        ],
    },
    encode: "simple",
    tokenize: "forward"
  });
```

Any ideas on where to look?

**Comment by krzko** [comment]
No issues in the console, and `var index = new FlexSearch();` seems to be instantiating just fine.

**Comment by h-enk** [comment, marked as answer]
@krzko, does it work when you use the default settings? So, when you remove:

```js
..
    encode: "simple",
    tokenize: "forward"
..
```

And, could you provide a link to your repo?

**Reply by krzko** [reply]
Just tried it without those settings same problem.

Sorry then repo is in a private org, cant share it out. Any particular files you want me to paste?

**Reply by krzko** [reply]
When I perform a search, the URL changes from `http://localhost:1313` to `http://localhost:1313/?` only. Is that right? No params?

**Reply by h-enk** [reply]
> When I perform a search, the URL changes from http://localhost:1313 to http://localhost:1313/? only. Is that right?

With an empty search or with no suggestion selected, that's the expected behavior.

**Reply by h-enk** [reply]
> Sorry then repo is in a private org, cant share it out. Any particular files you want me to paste?

FlexSearch works out of the box. It is hard to help when I do not know what files you changed, and if you're using the starter or child theme.

**Reply by h-enk** [reply]
https://github.com/h-enk/doks/issues/268?

**Reply by krzko** [reply]
We've not deployed the site as of yet, just doing all testing on http://localhost:1313, will that impact the search functionality?

Is there a specific change that needs to happy for local dev envs?

**Reply by h-enk** [reply]
> will that impact the search functionality?

No, it just works out of the box.

**Reply by krzko** [reply]
I've temporarily uploaded a cut down repo to here - https://github.com/krzko/doks-test, if you can have a look what could be wrong with it?

**Reply by h-enk** [reply]
> I've temporarily uploaded a cut down repo to here

That helped.

Change line 86 of `./assets/js/index.js` from:

```js
{{ range $index, $page := (where .Site.Pages "Section" "docs") -}}
```
to: 

```js
{{ range $index, $page := (where .Site.Pages "Section" "runbooks") -}}
```
And, now you can search in the playbook section:

![Snag_1366a00](https://user-images.githubusercontent.com/3902872/121008853-bd996a00-c793-11eb-9842-8847f2f6f10d.png)

I realize this could be a parameter (one or more sections to include ) to be set in `params.toml` — enhancement for Doks v0.3

**Reply by krzko** [reply]
Thanks so much, that worked perfectly. Legend. Thank you.
